Source/CSharpEssentials/Extensions.cs

```diff
@@ -168,9 +170,36 @@ private static bool IsGeneratedCode(string filePath)
             return false;
         }
 
-        public static bool IsGeneratedCode(this SyntaxTree tree)
+        public static bool IsGeneratedCode(this SyntaxNode node)
         {
-            return IsGeneratedCode(tree.FilePath);
+            return IsInGeneratedCodeContext(node)
+                || IsGeneratedCode(node.SyntaxTree.FilePath);
+        }
+
+        private static bool IsInGeneratedCodeContext(this SyntaxNode node)
+        {
+            do
+            {
+                var method = node as MethodDeclarationSyntax;
+                if (method != null && ContainsGeneratedCodeAttribute(method.AttributeLists))
+                    return true;
+
+                var classDecleration = node as ClassDeclarationSyntax;
+                if(classDecleration != null && ContainsGeneratedCodeAttribute(classDecleration.AttributeLists))
+                    return true;
+
+                node = node.Parent;
+            }
+            while (node != null);
+
+            return false;
+        }
+
+        private static bool ContainsGeneratedCodeAttribute(IEnumerable<AttributeListSyntax> attributes)
+        {
+            return attributes
+                .SelectMany(x => x.Attributes)
+                .Any(x => x.ToFullString().Contains("GeneratedCode"));
```

Comment:
This seems a bit too relaxed to me. Isn't "System.CodeDom.Compiler.GeneratedCode" the attribute you're looking for? Wouldn't it be better to find the ISymbol for that particular attribute type and then verify that one of the attributes binds to it?

Additionally, you're comparing "GeneratedCode" against the entire string of the attribute, including parameters and even comments and whitespace (that's what ```ToFullString()``` returns). So, this method would return true for this:

[NotGenerated /*GeneratedCode*/]

:smile: